features/storage/kvstore/filesystemstore/FileSystemStore.cpp

```diff
@@ -0,0 +1,647 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "FileSystemStore.h"
+#include "Dir.h"
+#include "File.h"
+#include "BlockDevice.h"
+
+#include "mbed_trace.h"
+#define TRACE_GROUP "FSST"
+
+using namespace mbed;
+
+
+FileSystemStore::FileSystemStore(/*size_t max_keys,*/ FileSystem *fs) : _fs(fs), _max_keys(FSST_MAX_KEYS), _num_keys(0),
+    _is_initialized(false)
+{
+    memset(_cfg_fs_path, 0, FSST_PATH_NAME_SIZE + 1);
+    strncpy(_cfg_fs_path, "$fsst$", FSST_PATH_NAME_SIZE);
+    _cfg_fs_path[FSST_PATH_NAME_SIZE] = '\0';
+    memset(_full_path_key, 0, FSST_PATH_NAME_SIZE + KVStore::MAX_KEY_SIZE + 1);
+    strncpy(_full_path_key, _cfg_fs_path, FSST_PATH_NAME_SIZE);
+    strcat(_full_path_key, "/");
+    _cur_inc_data_size = 0;
+}
+
+// Initialization and reset
+int FileSystemStore::init()
+{
+    int status = FSST_ERROR_OK;
+
+    printf("FileSystemStore Init Enter\n");
+    _mutex.lock();
+
+    Dir kv_dir;
```

Comment:
works fine with root.
(As Chris wrote it will save a bit of string manipulation.. I ran a small test and it would save  something like 200 bytes of code text) - cant see the major importance of it..